metrics-core/src/main/java/com/codahale/metrics/WeightedSnapshot.java

```diff
@@ -38,6 +38,12 @@ public WeightedSample(long value, double weight) {
      * @param values    an unordered set of values in the reservoir
      */
     public WeightedSnapshot(Collection<WeightedSample> values) {
+        if (values.size()==1 && Double.compare(values.iterator().next().weight,0D)==0) {
```

Comment:
Hi thanks, yes I will look into that today. I agree here it isn't the best to solve this problem.